Modified analytics

sanketbansal opened the pull request:
<!--
(Thanks for sending a pull request! .)
-->

<!--The PR title should start with "Fix #bugnum: " (if applicable), followed by a clear one-line present-tense summary of the changes introduced in the PR. For example: "Fix #bugnum: Introduce the first version of the collection editor." -->

<!-- Add issue number here. If you do not solve the issue entirely, please change the message e.g. "Addresses #IssueNumber -->


#### Changes proposed in this pull request:

<!-- Changes: Add here what changes were made in this issue and if possible provide links. -->
- Added host analytics page for current version of EvalAI.

<!-- Demo Link: Add here the link where you changes can be seen. -->
- Link to live demo: http://pr-186-evalai.surge.sh <!-- Replace XXX with your PR no: -->

<!-- Screenshots for the change: Add here the screenshot of the fix. -->
![Screenshot from 2019-07-21 16-21-54](https://user-images.githubusercontent.com/17106489/61596603-b84cc980-abd3-11e9-9860-3f0b5c2a759a.png)

sanketbansal committed:
Ported image assets

Ported the image assets from angularJs application into 'image-port' directory

sanketbansal committed:
Added styles from angularjs

Ported styles from angularjs Application inside style-port folder

sanketbansal committed:
Modified footer component

Added github fork and star button using github api

sanketbansal committed:
Modified auth service

Changes made in auth services to integrate angularJS authentication features

sanketbansal committed:
Changes in app component

Modified app component to show loader according to authService 'isLoader' variable.

sanketbansal committed:
Modified app loader component

app loader component modified to accompany AngularJS application UI.

sanketbansal committed:
Modified angular.json file

Added main.scss file of ported style in the angular.json file to compile it at the time of building project.

sanketbansal committed:
Removed routing from auth service

Routing to 'auth/login' has been removed

sanketbansal committed:
Modified loading component testBed

Added auth,global.api,endpoint services and httpclientmodule.

sanketbansal committed:
Added side bar component

Added side bar as utility component for the side bar in the Dashboard UI

sanketbansal committed:
Added dash footer component

Dash footer added for improving dashboard UI

sanketbansal committed:
changes in style files

Added font variables in ported variable file
Added font style in ported main file.

sanketbansal committed:
Modified footer

Made the footer UI similar to the angularjs application one.

sanketbansal committed:
Added simple header component

simple header added to modify the UI of the header in the dashboard

sanketbansal committed:
Modifications in simple header component

Added color property for signup button
Added styles for login button

sanketbansal committed:
Modified app component

Removed ngIf  condition from app-loading component to resolve the error:
ExpressionChangedAfterItHasBeenCheckedError

sanketbansal committed:
Added routing for analytics pages

Routing path has been added for analytics pages inside routing module file.

sanketbansal committed:
Added Analytics component

Analytics component page has been added with the UI similar to current evalAi version..

sanketbansal committed:
Added host analytics pages

Host analytics component has been added as a child component of the analytics component. To show analysis related to the host challenges and teams.

sanketbansal committed:
Added feature to start stop loader

Added methods which are used by the components to start and stop the loader.

sanketbansal committed:
Fixed linting issues

Added missing whitespace in analytics.component file

sanketbansal committed:
Moved host-analytics styles

Styles related to host-analytics moved from 'styles-port/modules/analytics.scss'

sanketbansal committed:
Fixed bug in Host analytics component html

Added ngModel directive to select challenge for the hosted challenges.

sanketbansal committed:
Fixed bugs in host analytics component

Fixed download api url.
Added loader UI while fetching data.
Changed challengelistcount after fetching challenges.
Changed logic to show the fetched last submission date

sanketbansal committed:
Resolved rebase bugs

Added proper json standards in angular.json and app.module file

sanketbansal committed:
Removed unused styles and assets

removed unused styles and assets
Refactored required files after removing styles and assets

sanketbansal committed:
Added styles

Added styles in styles-port folder

sanketbansal committed:
Added styles

Styles added into their specific component

sanketbansal committed:
Added styles for loading component

Added styles in base.scsss and variable.scss reqiored by loading component
Added styles in loading.component.scss file

sanketbansal committed:
Modifed project configuration

Added base style files globally in angular.json.
Removed duplicate imports
Removed main.scss file from styles-port folder

sanketbansal committed:
Fixed bug in simple header component

Added styles to resolve  the UI bug.

sanketbansal committed:
Changes made in analytics component

Added styles to fix the UI bug
Removed unused variables.
corrected the error in using cuuret challenge variable in UI.

sanketbansal committed:
Changes in global service

Added method and variable to show loading msg

sanketbansal committed:
changes in auth service

Removed unused variables

sanketbansal committed:
Changes in host-analytics component

Added start, stop loader method of globalService into hsot anlytcs component for app loading

sanketbansal committed:
Fixed bug in app module

Minor comma bug fixed in app module

sanketbansal committed:
Configuration changes

Removed styles from styles-port folder
Removed basic.sccs from angular.json decalarition

sanketbansal committed:
Removed unused components

Dash footer and Simple header componet has been removed

sanketbansal committed:
Changes in host analytics

Added font weight from base.scss in host analytic html

lunayach commented:
@sanketbansal pass the build.

sanketbansal committed:
Changes in host analytics

Removed the use of app simple header and dash-footer

sanketbansal committed:
Chabges in app loading

Fixed bug in app loader

sanketbansal committed:
Changes in app loader

Added loader title

sanketbansal committed:
Fixed test cases in anlytics compnent

Removed dash footer component from test bed configuration.
Removed simple header from test bed configuration.
Added header static component  into the test bed configuration.

sanketbansal committed:
Removed logs and comments

Logs and comments removed from anlaytics page

sanketbansal committed:
Changes in auth service

Removed unused changes from auth service file

sanketbansal committed:
Changes in app module

Changed the place of sidebar component import.

sanketbansal committed:
Changes in app component

Changed authservice scope to private

lunayach commented on src/app/components/analytics/host-analytics/host-analytics.component.ts:
```diff
@@ -0,0 +1,186 @@
+
+  downloadChallengeParticipantTeams() {
+
+    this.apiService.getUrl(`analytics/challenges/${this.challengeId}/download_all_participants/`, false)
```
Define the URL as a constant variable above.

lunayach commented on src/app/components/analytics/host-analytics/host-analytics.component.ts:
```diff
@@ -0,0 +1,186 @@
+      this.isTeamSelected = true;
+
+      this.globalService.startLoader(`Fetching Team count for challenge ${this.challengeId}`);
+      this.apiService.getUrl('analytics/challenge/' + this.challengeId + '/team/count').subscribe(
```
Here, as well.

lunayach commented on src/app/components/analytics/host-analytics/host-analytics.component.ts:
```diff
@@ -0,0 +1,186 @@
+          for (let phaseCount = 0; phaseCount < this.currentPhase.length; phaseCount++) {
+            challengePhaseId.push(this.currentPhase[phaseCount].id);
+
+            // tslint:disable-next-line:max-line-length
```
Avoid these kinds of hacks :slightly_smiling_face:.

lunayach commented on src/app/components/analytics/host-analytics/host-analytics.component.ts:
```diff
@@ -0,0 +1,186 @@
+          }
+
+          for (let phaseCount = 0; phaseCount < this.currentPhase.length; phaseCount++) {
+            // challengePhaseId.push(this.currentPhase[phaseCount].id);
```
Remove both of these.

lunayach commented on src/app/components/analytics/host-analytics/host-analytics.component.ts:
```diff
@@ -0,0 +1,186 @@
+      for (let i = 0; i < this.challengeList.length; i++) {
+        // tslint:disable-next-line:triple-equals
+        if (this.challengeList[i]['id'] == this.challengeId) {
+          this.currentChallengeDetails = this.challengeList[i];
```
In **JS**, we should always use **===** to check for equality. If a type conversion is needed, do it before this.

sanketbansal replied:
that has been done

lunayach commented on src/app/components/analytics/host-analytics/host-analytics.component.scss:
```diff
@@ -0,0 +1,60 @@
+  display:flex;
+  flex-direction: row;
+  .show-count {
+    margin-left: 5px !important;
```
Do we really need **!important** here? (Also, at other places.)

sanketbansal replied:
Yeah, Also using **!important** inside component specific css is toatally safe. It do not hinder with other css

lunayach replied:
It does not hinder but overrides, that might cause sometimes unexpected behavior. I meant that we should only use this when we really need it.

sanketbansal replied:
It overrides for component specific html only that's why it's safe

lunayach commented on src/app/components/analytics/host-analytics/host-analytics.component.html:
```diff
@@ -0,0 +1,117 @@
+                    </span>
+          <span class="show-count">
+                        {{totalChallengeTeams}}
+                    </span>
```
```suggestion
   </span>
```

lunayach commented on src/app/components/analytics/host-analytics/host-analytics.component.html:
```diff
@@ -0,0 +1,117 @@
+          <div class="col s12 m4 l2">
+            <div class="image-container analytics-image">
+              <img *ngIf="currentChallengeDetails['image']!==null" [src]='currentChallengeDetails["image"]'>
+              <img *ngIf="currentChallengeDetails['image'] == null" src='assets/images/404.png'>
```
```suggestion
              <img *ngIf="currentChallengeDetails['image'] === null" src='assets/images/404.png'>
```

sanketbansal committed:
Changes in host analytics

Added analytics url inside endpoint service
Modified host-analytics to use analytics  url from endpoints service.
Added variable for route path

lunayach commented on src/app/components/analytics/host-analytics/host-analytics.component.html:
```diff
@@ -0,0 +1,117 @@
+                  Latest Submission At
+                </div>
+                <p *ngIf="lastSubmissionTime[item.id]" class="no-top-para mb-5">
+                  {{lastSubmissionTime[item.id] | date:'medium'}}
```
Here, when you have no submissions, `lastSubmissionTime[item.id]` is returned as _**You dont have any submissions in this challenge phase!**_. So, eventually, it enters under the `*ngIf` but then leads to a console error as that string is incompatible with the `Date` pipe.

sanketbansal replied:
yeah, I think I have handled that case in the logic. i.e **lastsubmissionTime[item.id]** will either be **date string** or **undefined**

sanketbansal replied:
are you getting error on your machine regarding that?

sanketbansal replied:
resolved?

lunayach replied:
Can you try when no. of submissions in a phase are zero? Because I seem to be getting the above error.

sanketbansal replied:
I have tried that for zero participants also. it shows none as expected

lunayach replied:
This, when I again tried in a different machine.

![pipe_error](https://user-images.githubusercontent.com/21227893/61628956-3aadba00-acb6-11e9-884e-14bb23ec4f0f.png)

lunayach replied:
Thanks for resolving this. The error is gone now.

lunayach commented on src/app/components/analytics/host-analytics/host-analytics.component.html:
```diff
@@ -0,0 +1,117 @@
+          </label>
+        </li>
+      </ul>
+      <div class="mt-30 btn ev-btn-dark waves-effect waves-dark grad-btn grad-btn-dark fs-14 show-analytics-btn" [hidden]="challengeListCount === 0" (click)="showChallengeAnalysis()">Show Analysis &nbsp;<i class="fa fa-chevron-right"></i> </div>
```
I saw that `[hidden]="challengeListCount === 0"` was not working when `challengeListCount` was zero. Either use `*ngIf="challengeListCount !== 0"` or you can also disable it for the above condition (like in current version).

sanketbansal replied:
done

lunayach reviewed (commented):
As the screenshot in https://github.com/Cloud-CV/EvalAI-ngx/pull/186#issue-299662720 suggests, the text in **Total Submissions**, **Participant Teams**, seems misaligned. Can that be taken care of in this PR, or your future PR would fix that?

Otherwise, it's working!

sanketbansal commented:
> As the screenshot in [#186 (comment)](https://github.com/Cloud-CV/EvalAI-ngx/pull/186#issue-299662720) suggests, the text in **Total Submissions**, **Participant Teams**, seems misaligned. Can that be taken care of in this PR, or your future PR would fix that?

I think it's like that in current UI

sanketbansal commented:
> As the screenshot in [#186 (comment)](https://github.com/Cloud-CV/EvalAI-ngx/pull/186#issue-299662720) suggests, the text in **Total Submissions**, **Participant Teams**, seems misaligned. Can that be taken care of in this PR, or your future PR would fix that?

Done

lunayach reviewed (approved):
Now it looks good to me. LGTM! @RishabhJain2018

RishabhJain2018 commented on src/app/components/analytics/host-analytics/host-analytics.component.ts:
```diff
@@ -0,0 +1,193 @@
+            this.apiService.getUrl(PATH_API)
+              .subscribe(
+                (res) => {
+                  console.log(res);
```
Please remove this console statement.

sanketbansal replied:
done

RishabhJain2018 reviewed (changes requested):
Minor Comment @sanketbansal. Rest looks good! 🎉
